src/cljs/main/broadfcui/common/table/paginator.cljs

```diff
@@ -0,0 +1,92 @@
+(ns broadfcui.common.table.paginator
+  (:require
+    [inflections.core :refer [pluralize]]
+    [broadfcui.common.icons :as icons]
+    [broadfcui.common.style :as style]
+    [broadfcui.utils :as utils]
+    ))
+
+
+(defn- create-page-range [current-page total-pages]
+  (cond
+    (<= total-pages 5) (range 1 (inc total-pages))
+    (<= current-page 3) (range 1 6)
+    (>= current-page (- total-pages 2)) (range (- total-pages 4) (inc total-pages))
+    :else (range (- current-page 2) (+ current-page 3))))
+
+
+(defn- container [narrow? child align]
+  (if narrow?
+    [:div {:style {:margin "0.25em 0"}} child]
+    [:div {:style {:width "33%" :textAlign align}} child]))
+
+
+(defn- view-component [{:keys [page-number rows-per-page filtered-count total-count]}]
+  (let [right-num (min filtered-count (* page-number rows-per-page))
+        left-num (if (zero? right-num) 0 (inc (* (dec page-number) rows-per-page)))]
+    [:div {:style {:display "inline-flex"}}
+     [:b {:style {:marginRight "1ex"}} (str left-num " - " right-num)]
+     (str "of " (pluralize filtered-count " result")
+          (when-not (= filtered-count total-count)
+            (str " (filtered from " total-count " total)")))]))
+
+
+(defn- page-component [{:keys [filtered-count rows-per-page page-number page-selected]}]
+  (let [num-pages (js/Math.ceil (/ filtered-count rows-per-page))
+        allow-prev (> page-number 1)
+        allow-next (< page-number num-pages)]
+    (style/create-unselectable
+     :div {:style {:display "inline-flex" :alignItems "baseline"}}
+     [:div {:style {:display "inline-flex" :alignItems "baseline"
+                    :padding "0em 0.9em"
+                    :color (if allow-prev
+                             (:button-primary style/colors)
+                             (:border-light style/colors))
+                    :cursor (when allow-prev "pointer")}
+            :onClick (when allow-prev #(page-selected (dec page-number)))}
+      (icons/icon {:style {:alignSelf "center" :paddingRight "0.5rem"}} :angle-left)
+      "Prev"]
+     [:span {:style {:whiteSpace "nowrap"}}
+      (map (fn [n]
+             (let [selected? (= n page-number)]
+               [:div {:style {:textAlign "center"
+                              :paddingTop 5 :display "inline-block" :width 29 :height 24
```

Comment:
Pixel pushing to make those divs square.  I would love to know a better way to do this though!